server/src/com/cloud/network/element/VirtualRouterElement.java

```diff
@@ -227,7 +227,7 @@ public boolean implement(final Network network, final NetworkOffering offering,
         final List<DomainRouterVO> routers = routerDeploymentDefinition.deployVirtualRouter();
 
         int routerCounts = 1;
-        if (offering.getRedundantRouter()) {
+        if (offering.getRedundantRouter() || network.isRollingRestart()) {
```

Comment:
neat solution

Reply:
Unlikely we'll have more than 2 routers for redundancy. However, the code has an internal limit of 5 (after which it logs warning/errors that a network has more than 5 routers, which can happen in some edge cases).